data/test/scenarios/wml_tests/UnitsWML/zero_experience.cfg

```diff
@@ -0,0 +1,80 @@
+# wmllint: no translatables
+
+#####
+# API(s) being tested: [unit] max_experience=0
+##
+# Actions:
+# Change Alice's and Bob's max_experience to 0. Delete Bob's 'Orcish Warrior' advancement but not Alice's. Harm bob with 1 hitpoint to be sure that he have some damage. They then fight each other.
+##
+# Expected end state:
+# Alice levels up. Bob doesn't, and keeps his previous health.
+#####
+{GENERIC_UNIT_TEST "zero_experience" (
+    [event]
+        name=prestart
+        [modify_unit]
+            [filter]
+                id=alice,bob
+            [/filter]
+            [effect]
+                apply_to=max_experience
+                set=0
+            [/effect]
+        [/modify_unit]
+        [modify_unit]
+            [filter]
+                id=bob
+            [/filter]
+            [object]
+                [effect]
+                    apply_to=remove_advancement
+                    types=Orcish Warrior
+                [/effect]
+            [/object]
+        [/modify_unit]
+        [harm_unit]
+            [filter]
+                id=bob
+            [/filter]
+            amount=1
+        [/harm_unit]
+    [/event]
+    [event]
+        name=start
+        [do_command]
+            [move]
+                x=13,12,11,10,9,8
+                y=3,3,3,3,3,3
+            [/move]
+            [attack]
+                weapon=0
+                defender_weapon=0
+                [source]
+                    x,y=7,3
+                [/source]
+                [destination]
+                    x,y=8,3
+                [/destination]
+            [/attack]
+        [/do_command]
+    [/event]
+    [event]
+        name=turn 1
+        [store_unit]
+            [filter]
+                id=alice
+            [/filter]
+            variable=alice
+        [/store_unit]
+        [store_unit]
+            [filter]
+                id=bob
+            [/filter]
+            variable=bob
+        [/store_unit]
+        {ASSERT ({VARIABLE_CONDITIONAL bob.hitpoints numerical_not_equals max_hitpoints})}
```

Comment:
Should be `$max_hitpoints`.

Reply:
$bob.max_hitpoints, yes
An oversight. Since it didn't give an error, I didn't realize